tests/puzzle/test_solver.py

```diff
@@ -793,6 +793,51 @@ def test_solver_finds_extras_next_to_non_extras(
     )
 
 
+def test_solver_merge_extras_into_base_package_multiple_repos_fixes_5727(
+    solver: Solver, repo: Repository, pool: Pool, package: ProjectPackage
+):
+    package.add_dependency(
+        Factory.create_dependency(
+            "A",
+            {
+                "version": "*",
+                "source": "legacy",
+            },
+        )
+    )
+    package.add_dependency(
+        Factory.create_dependency(
+            "B",
+            {
+                "version": "*",
+            },
+        )
+    )
```

Comment:
Can we make this a bit more dense? (Just remove the trailing commas and let pre-commit do its work.)